scipy/linalg/flapack_pos_def_tri.pyf.src

```diff
@@ -12,3 +12,73 @@ subroutine <prefix>ptsv(n, nrhs, d, e, b, info)
     integer intent(out) :: info
 
 end subroutine <prefix>ptsv
+
+
+subroutine <prefix>pttrf(n, d, e, info)
+    ! d, e, info  = pttrf(d, e, overwrite_d=0, overwrite_e=0)
+    !
+    ! ?PTTRF computes the L*D*L**T factorization of a symmetric positive
+    ! definite tridiagonal matrix A.  The factorization may also be regarded as ! having the form A = U**T*D*U.
+
+    callstatement (*f2py_func)(&n, d, e, &info)
+    callprotoargument int*, <ctypereal>*, <ctype>*, int*
+
+    integer intent(hide), depend(d), check(n > 1) :: n = len(d)
+    <ftypereal> intent(in, out, copy), dimension(n) :: d
+    <ftype> intent(in, out, copy), dimension(n - 1), depend(n) :: e
+    integer intent(out) :: info
+
+end subroutine <prefix>pttrf
+
+
+subroutine <prefix2>pttrs(n, nrhs, d, e, b, ldb, info)
+    ! x, info  = pttrs(d, e, b, overwrite_b=0)
+    !
+    ! DPTTRS solves a tridiagonal system of the form
+    ! 
+    !     A * X = B
+    !
+    ! using the L*D*L**T factorization of A computed by DPTTRF.  D is a
+    ! diagonal matrix specified in the vector D, L is a unit bidiagonal
+    ! matrix whose subdiagonal is specified in the vector E, and X and B
+    ! are N by NRHS matrices.
+
+    callstatement (*f2py_func)(&n, &nrhs, d, e, b, &ldb, &info)
+    callprotoargument int*, int*, <ctype2>*, <ctype2>*, <ctype2>*, int*, int*
+
+    integer intent(hide), depend(d), check(n > 1) :: n = len(d)
+    <ftype2> intent(in), dimension(n) :: d
+    <ftype2> intent(in), depend(n), dimension(n - 1) :: e
+    <ftype2> intent(in, out, copy, out=x), dimension(ldb, nrhs) :: b
+    integer intent(hide), depend(b) :: ldb = MAX(1, shape(b, 0))
+    integer intent(hide), depend(b)  :: nrhs = MAX(1, shape(b, 1))
+    integer intent(out) :: info
+
+end subroutine<prefix2>pttrs
+
+
+subroutine <prefix2c>pttrs(uplo, n, nrhs, d, e, b, ldb, info)
+    ! x, info  = pttrs(d, e, b, uplo='L', overwrite_b=0)
+    !
+    ! ?PTTRS solves a tridiagonal system of the form
+    !
+    !     A * X = B
+    !
+    ! using the L*D*L**T factorization of A computed by DPTTRF.  D is a
+    ! diagonal matrix specified in the vector D, L is a unit bidiagonal
+    ! matrix whose subdiagonal is specified in the vector E, and X and B
+    ! are N by NRHS matrices.
+
+    callstatement (*f2py_func)(uplo, &n, &nrhs, d, e, b, &ldb, &info)
+    callprotoargument char*, int*, int*, <ctype2>*, <ctype2c>*, <ctype2c>*, int*, int*
+
+    character optional, check(*uplo=='U'||*uplo=='L') :: uplo = 'L'
```

Comment:
The uplo stuff is generally done via `lower=1` and `lower=0` in our wrappers. I don't like the inconsistencies either of having both uplo and lower but it is what it is. Also numbers are tiny bit easier to enter. So better we keep the lower keyword instead

Reply:
OK, I think I fixed it... would you check me? Although default `lower=0` is more common for other routines, I kept the default at `lower=1` because L D L' is more familiar than U' D U.

Reply:
This was my argument too when I was adding LDL decomposition but somebody pointed out that overwhelmingly we have `lower=0` in all wrappers and peope chimed in for consistency. So probably we should follow that.

Reply:
OK.